homeassistant/components/media_player/sisyphus.py

```diff
@@ -0,0 +1,202 @@
+
+    def __init__(self, name, host, table):
+        """
+        Constructor.
+
+        :param name: name of the table
+        :param host: hostname or ip address
+        :param table: sisyphus-control Table object
+        """
+        self._name = name
+        self._host = host
+        self._table = table
+        self._initialized = False
+
+    def update(self):
+        """Lazily initializes the table."""
+        if not self._initialized:
+            # We wait until update before adding the listener because
+            # otherwise there's a race condition by which this entity
+            # might not have had its hass field set, and thus
+            # the schedule_update_ha_state call will fail
+            self._table.add_listener(
+                lambda: self.schedule_update_ha_state(False))
+            self._initialized = True
+
+    @property
+    def name(self):
+        """Return the name of the table."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the current state of the table; sleeping maps to off."""
+        if self._table.state in ["homing", "playing"]:
+            return STATE_PLAYING
+        elif self._table.state == "paused":
+            if self._table.is_sleeping:
+                return STATE_OFF
+
+            return STATE_PAUSED
+        elif self._table.state == "waiting":
+            return STATE_IDLE
+
+        return None
+
+    @property
+    def volume_level(self):
+        """Return the current playback speed (0..1)."""
+        return self._table.speed
+
+    @property
+    def shuffle(self):
+        """Return True if the current playlist is in shuffle mode."""
+        return self._table.is_shuffle
+
+    async def async_set_shuffle(self, shuffle):
+        """
+        Change the shuffle mode of the current playlist.
+
+        :param shuffle: True to shuffle, False not to
+        """
+        await self._table.set_shuffle(shuffle)
+
+    @property
+    def media_playlist(self):
+        """Return the name of the current playlist."""
+        return self._table.active_playlist.name \
+            if self._table.active_playlist \
+            else None
+
+    @property
+    def media_title(self):
+        """Return the title of the current track."""
+        return self._table.active_track.name \
+            if self._table.active_track \
+            else None
+
+    @property
+    def media_content_type(self):
+        """Return the content type currently playing; i.e. a Sisyphus track."""
+        return MEDIA_TYPE_TRACK
+
+    @property
+    def media_content_id(self):
+        """Return the track ID of the current track."""
+        return self._table.active_track.id \
+            if self._table.active_track \
+            else None
+
+    @property
+    def supported_features(self):
+        """Return the features supported by this table."""
+        return SUPPORT_VOLUME_MUTE \
```

Comment:
Break out the features into a constant put at the module level and return that here.